04-Multitasking/context_switch.S

```diff
@@ -19,14 +19,32 @@ activate:
 	mrs ip, psr
 	push {r4, r5, r6, r7, r8, r9, r10, r11, ip, lr}
 
-	/* switch to process stack */
-	msr psp, r0
-	mov r0, #3
-	msr control, r0
-	isb
 
 	/* load user state */
-	pop {r4, r5, r6, r7, r8, r9, r10, r11, lr}
+	ldmia r0!, {r4, r5, r6, r7, r8, r9, r10, r11, lr}
+
+	msr psp, r0
+
+	/* check the situaion and determine the path to return */
```

Comment:
Why would you mention it as the path to return? I don't really think it is a path. Instead, it is transition.